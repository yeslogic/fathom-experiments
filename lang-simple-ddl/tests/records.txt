type Point {
  x : Int64;
  y : Int64;
}

def test-point-1 : Point := {
  x := 1;
  y := 2;
};

def test-point-2 :=
  { x := 1; y := 2 } : Point;

type A-long-record {
  foo : Int64;
  bar : Int64;
  baz : Int64;
  bananas : Int64;
}

def test-long-record : A-long-record := {
  foo := 1000000;
  bar := 1000000;
  baz := 1000000;
  bananas := 1000000;
};
