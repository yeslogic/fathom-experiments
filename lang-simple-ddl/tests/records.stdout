fn read_byte(input: &[u8], pos: &mut usize) -> Result<i64, ()> {
    let byte = input.get(*pos).ok_or(())?;
    *pos +=1;
    Ok(i64::from(*byte))
}

struct Point {
    x: i64,
    y: i64,
}

const TEST_POINT_1: Point = Point { x: 1, y: 2 };

const TEST_POINT_2: Point = Point { x: 1, y: 2 };
